packages/client/src/v2-events/features/events/AdvancedSearch/TabSearch.tsx

```diff
@@ -107,11 +110,16 @@ export function TabSearch({
     Record<string, FieldValue>
   >(fieldValues ?? {})
 
-  const hasEnoughParams = Object.entries(formValues).length > 0
   const navigate = useNavigate()
 
+  const prevEventId = React.useRef(currentEvent.id)
+
   React.useEffect(() => {
-    setFormValues({})
+    // only reset formValues if another event search tab is selected
+    if (prevEventId.current !== currentEvent.id) {
+      setFormValues({})
+      prevEventId.current = currentEvent.id
+    }
   }, [currentEvent])
```

Comment:
Could we just do this? I'm not sure if it works 100% similarly, so good to double check

```typescript
  React.useEffect(() => {
    setFormValues({})
  }, [currentEvent.id])
```

Reply:
So, what I found was that, the `props.fieldValues` is used to set the form initialValue, and the `formValues` contains the current formValues.

When there is no check in the useEffect, when we are redirecting from `SearchResult` page to AdvancedSearch, on component mount, the useEffect runs, and it directly set `formValues` to `{}`. Which results in unpopulating the search fields, where as, upon redirection, the search fields should be populated.